### PR TITLE
Button docs

### PR DESCRIPTION
**Pull request by indirectlylit**
# Component documentation

Adds documentation for library components:

* `KButton`
* `KRouterLink`
* `KExternalLink`
* `KButtonGroup`
* `KIconButton`

# Filter keywords

Adds a new 'keywords' property to the table of contents to allow filtering by related terms. For example:

```js
      new Page({
        path: '/KButton',
        title: 'KButton',
        isCode: true,
        keywords: ['button', 'link'],
      }),
```

# Cross-links to components

Adds a new `DocsLibraryLink` component to allow easy cross-linking. For example:

```vue
<DocsLibraryLink page="KExternalLink" /> is used to create standard <code>&lt;a&gt;</code> tags with an <code>href</code> attribute.
```

### CODE REVIEW

**Comment by sairina** [comment]
This is great! Two follow-ups, I believe: 

1. We may also want to update the README (for example to include `<DocsLibraryLink>`?)
2. While we're in the README, I think there also is a small typo in the README [here](https://github.com/learningequality/kolibri-design-system#running-the-documentation-locally) where it says 
```
cd design-system
yarn install 
```
and I think it should be `cd kolibri-design-system`

**Review by sairina — APPROVED** [review]
LGTM

**Review by nucleogenesis — DISMISSED** [review]
This PR is awesome.

One bug I've noted (possibly unrelated to this work) is that the filtering in the sidebar isn't working 100% correctly.

I can make a new issue if it's not related to this work.

image

**Comment by indirectlylit** [comment]
thanks for the feedback both. Ready for re-review.

**Review by sairina — APPROVED** [review]
Manually tested the issue that @nucleogenesis brought up, and the filtering with "button" seems to work now. LGTM!

File: docs/pages/buttons.vue [hunk 1]
```diff
@@ -102,58 +92,84 @@
...
       <ul>
-        <li>Right-aligned in modal pop-ups</li>
-        <li>Left-aligned in page forms</li>
-        <li>Global actions related to tables should be put above the table and right-aligned</li>
+        <li>Right-aligned at the bottom of modal pop-ups</li>
+        <li>Left-aligned a the bottom of page forms</li>
+        <li>Global actions related to full pages should be right-aligned and to the right of the page title on large screens, and left-aligned below the title on small screens. Use of <code>KGrid</code> is recommended</li>
       </ul>
+      <p>
+        Use the <DocsLibraryLink page="KButtonGroup" /> component to add specing between adjacent buttons.
```
Comment (jtamiace) [inline docs/pages/buttons.vue:109]:
```suggestion
        Use the <DocsLibraryLink page="KButtonGroup" /> component to add spacing between adjacent buttons.
```

Comment (indirectlylit) [reply docs/pages/buttons.vue:109]: lol thanks good catch. I have spec'ing on the brain

**Pull request merged**